platform-velocity/src/main/java/dev/hypera/chameleon/platform/velocity/event/VelocityListener.java

```
@@ -83,19 +82,22 @@ public void onPostLoginEvent(@NotNull PostLoginEvent event) {
      */
     @Subscribe
     public void onChatEvent(@NotNull PlayerChatEvent event) {
+        boolean immutable = event.getPlayer().getProtocolVersion().getProtocol() >= 760;
         UserChatEvent chameleonEvent = new UserChatEvent(
             this.chameleon.getUserManager().wrap(event.getPlayer()),
             event.getMessage(),
-            !event.getResult().isAllowed()
+            !event.getResult().isAllowed(),
+            immutable, immutable
         );
         this.chameleon.getEventBus().dispatch(chameleonEvent);
 
-        if (!event.getMessage().equals(chameleonEvent.getMessage()) &&
-            catchChatModification(event.getPlayer(), false)) {
+        if (immutable) {
+            return;
+        }
```

Shouldn't we log when an impossible modification has been made?

---

I am honestly not sure - I was debating whether it would be best to silently drop, log or throw an exception.
What are your thoughts?

---

Silently dropping could mean that developers would not be able to figure out why cancellation is not working.

---

Good point, I will add logging then